torch/_dynamo/variables/user_defined.py

```diff
@@ -1175,6 +1211,54 @@ def odict_getitem(self, tx: "InstructionTranslator", key):
         )(collections.OrderedDict.__getitem__(self.value, key.as_python_constant()))
 
 
+class FrozenDataClassVariable(UserDefinedObjectVariable):
```

Comment:
tbh I don't really know the best practices for when we decide to subclass from `UserDefinedObjectVariable` vs. a vanilla `VariableTracker`. But I guess it seems.. reasonable, since the object can have arbitrary fields?

Reply:
I see the old (unused) one that you deleted inherits from `ConstDictVariable` (`DataClassVariable(ConstDictVariable)`). would that make things any cleaner?

Reply:
Yeah, it wouldn't help, I think that was a bad idea tbh. It has a lot more functionality than dictionaries.  I inherited from UserDefinedObject because this object has arbitrary methods which we have to trace through. In the future this may need to be a mixin or something because this is going to be a common problem as we expand into new types of objects, for example subclassing builtin dictionaries would also require something similar to this.